src/content/docs/http-api/outputs/rtp.mdx

```diff
@@ -144,126 +144,24 @@ Root of a component tree/scene that should be rendered for the output. Use [`upd
-```tsx
-type VideoEncoderOptions =
-  | {
-      type: "ffmpeg_h264";
-      preset?:
-        | "ultrafast"
-        | "superfast"
-        | "veryfast"
-        | "faster"
-        | "fast"
-        | "medium"
-        | "slow"
-        | "slower"
-        | "veryslow"
-        | "placebo";
-      pixel_format?: "yuv420p" | "yuv422p" | "yuv444p";
-      ffmpeg_options?: Map<string, string>;
-    }
-  | {
-      type: "ffmpeg_vp8";
-      ffmpeg_options?: Map<string, string>;
-    }
-  | {
-      type: "ffmpeg_vp9";
-      pixel_format?: "yuv420p" | "yuv422p" | "yuv444p";
-      ffmpeg_options?: Map<string, string>;
-    }
-  | {
-      type: "vulkan_h264";
-      bitrate: { average_bitrate: number; max_bitrate: number; } | number;
-    };
-```
-</CollapsibleDetails>
-
-### Properties (type: "ffmpeg_h264")
-
-#### preset
-Video output encoder preset. Visit <a href='https://trac.ffmpeg.org/wiki/Encode/H.264#Preset' target='_blank'>FFmpeg docs</a> to learn more.
-
-- **Type**: `"ultrafast" | "superfast" | "veryfast" | "faster" | "fast" | "medium" | "slow" | "slower" | "veryslow" | "placebo"`
-- **Default value**: `fast`
-- **Supported values**:
-  - `ultrafast`
-  - `superfast`
-  - `veryfast`
-  - `faster`
-  - `fast`
-  - `medium`
-  - `slow`
-  - `slower`
-  - `veryslow`
-  - `placebo`
-
----
-
-#### pixel_format
-Encoder pixel format
-
-- **Type**: `"yuv420p" | "yuv422p" | "yuv444p"`
-- **Default value**: `yuv420p`
-- **Supported values**:
-  - `yuv420p`
-  - `yuv422p`
-  - `yuv444p`
-
----
-
-#### ffmpeg_options
-Raw FFmpeg encoder options. Visit <a href='https://ffmpeg.org/ffmpeg-codecs.html' target='_blank'>FFmpeg docs</a> to learn more.
-
-- **Type**: `Map<string, string>`
-
-### Properties (type: "ffmpeg_vp8")
-
-#### ffmpeg_options
-Raw FFmpeg encoder options. Visit <a href='https://ffmpeg.org/ffmpeg-codecs.html' target='_blank'>FFmpeg docs</a> to learn more.
-
-- **Type**: `Map<string, string>`
-
-### Properties (type: "ffmpeg_vp9")
-
-#### pixel_format
-Encoder pixel format
-
-- **Type**: `"yuv420p" | "yuv422p" | "yuv444p"`
-- **Default value**: `yuv420p`
-- **Supported values**:
-  - `yuv420p`
-  - `yuv422p`
-  - `yuv444p`
-
----
-
-#### ffmpeg_options
-Raw FFmpeg encoder options. Visit <a href='https://ffmpeg.org/ffmpeg-codecs.html' target='_blank'>FFmpeg docs</a> to learn more.
-
-- **Type**: `Map<string, string>`
-
-### Properties (type: “vulkan_h264”) <Badge text="Required feature: vk-video" variant="note" />
-Hardware encoder. Requires GPU that supports Vulkan Video encoding.
-
-#### bitrate
-Encoding bitrate. If not provided, bitrate is calculated based on resolution and framerate. For example at 1080p 30 FPS the average bitrate is 5000 kbit/s and max bitrate is 6250 kbit/s.
 
 <CollapsibleDetails summaryTitle='Type definitions' open>
+
 ```tsx
-type VulkanH264EncoderBitrate = { average_bitrate: number; max_bitrate: number; } | number;
+type VideoEncoderOptions = 
+  | ({ type: "ffmpeg_h264"; } & FfmpegH264EncoderOptions)
```

Comment:
Inconsistent 
both FfmpegH264EncoderOptions have field type and you are merging wit object that has type

Reply:
ok if we have that in other places you can leave it for now, but we need to think about some better approach for that